engine/execution/testutil/fixtures.go

```diff
@@ -659,12 +658,30 @@ func EntropyProviderFixture(source []byte) environment.EntropyProvider {
 // supports AtBlockID to return a snapshot mock.
 // The snapshot mock only supports RandomSource().
 // If input is nil, a random source fixture is generated.
-func ProtocolStateWithSourceFixture(source []byte) protocol.State {
+func ProtocolStateWithSourceFixture(source []byte) *protocolMock.State {
 	if source == nil {
 		source = unittest.SignatureFixture()
 	}
 	snapshot := &protocolMock.Snapshot{}
+	kvstore := &protocolMock.KVStoreReader{}
+	// TODO(mainnet27, #6773): remove GetProtocolStateVersion mock
```

Comment:
This is a follow up on your argument this morning, that we should continue to use the KV store version as a proxy for the protocol version. I am still struggling with fully comprehending what parts we will continue to need, vs which parts of the code are tied to the particular nature of the current change.

```suggestion
	// TODO(mainnet27, #6773): remove GetProtocolStateVersion mock, because this is only temporarily needed in the execution node to produce different chunk Data Packs depending on the protocol version 
```